op-node/rollup/derive/pipeline.go

```diff
@@ -11,7 +11,7 @@ import (
 )
 
 type L1Fetcher interface {
-	L1HeadBlockRef(ctx context.Context) (eth.L1BlockRef, error)
+	L1BlockRefByLabel(ctx context.Context, label eth.BlockLabel) (eth.L1BlockRef, error)
```

Comment:
Perhaps the rpc type from upstream is useful here

Reply:
Will open a ticket to consolidate `ByLabel`/`ByNumber`/`ByHash`/`ByRPCNumber`. There are 4 query param types I'm aware of, we can just generalize it, and support upstream param types as part of that.